src/engine/drpc_client.c

```diff
@@ -20,7 +22,140 @@
 /** dRPC client context */
 struct drpc *dss_drpc_ctx;
 
-/* Notify daos_server that we are ready (e.g., to receive dRPC requests). */
+/* dRPC socket (full, not directory-only) path */
+static char *dss_drpc_path;
+
+struct dss_drpc_thread_arg {
+	int32_t		  cta_module;
+	int32_t		  cta_method;
+	int		  cta_flags;
+	void		 *cta_req;
+	size_t		  cta_req_size;
+	Drpc__Response	**cta_resp;
+	volatile int	  cta_rc;
```

Comment:
(style)  Missing a blank line after declarations
(style)  Use of volatile is usually wrong: see Documentation/volatile-considered-harmful.txt